cassandra-launcher/src/main/scala/akka/persistence/cassandra/testkit/CassandraLauncher.scala

```diff
@@ -82,6 +82,14 @@ object CassandraLauncher {
     port
   }
 
+  @tailrec
+  private def freePort(usedPort: Int): Int = {
+    val port = freePort()
+    // bad luck? try again
+    if (port == usedPort) freePort(usedPort)
+    else port
+  }
```

Comment:
Unfortunately, `freePort` is public API, so instead, I'm adding an overloaded method, but private. 

Should deprecate `freePort`? I don't think we should have this as public API.

Reply:
👍

Reply:
Worse case freePort is not random but the OS insist on using the same over and over here, resulting in infinite loop.

I think it's better we we pick two ports at the same time, before closing them.

There is a `SocketUtil` in Akka for this, but that is in testkit and we don't want that dependency from the launcher.

Reply:
👍, I will reserve both together while keeping the sockets open and bound during the selection.